File: Day.py
#  File: Day.py

#  Description: The program will print out the day with the inputs of the date and the year. 

#  Student Name: Chengxi Li

#  Student UT EID: cl29385

#  Course Name: CS 303E

#  Unique Number: 53466

#  Date Created: Feb 16 2014

#  Date Last Modified: Feb 17 2014

def main():
  x = int(input("Enter year: "))
  y = int(input("Enter month: "))
  b = int(input("Enter day: "))
  d = x // 100
  if (y == 1 or y == 2):
    a = y + 10
    c = (x - 1 ) % 100
    d = x // 100 - 1
  else:
    a = y - 2
    c = x % 100
    d = x // 100
  w = (13 * a - 1 ) // 5 
  x = c // 4 
  y = d // 4 
  z = w + x + y + b + c - 2 * d 
  if ((z % 7) >= 0):
    r = z % 7
  else:
    r = (r + 7) % 7
  print("")
#  print("The day is", ("Monday." if r == 1 else \
#                      ("Tuesday." if r == 2 else \
#                      ("Wednesday." if r == 3 else \
#                      ("Thursday." if r == 4 else \
#                      ("Friday." if r == 5 else \
#                      ("Saturday." if r == 6 else \
#                      ("Sunday."))))))))
  if (r == 1):
    day = "Monday."
  elif (r == 2):
    day = "Tuesday."
  elif (r == 3):
    day = "Wednesday."
  elif (r == 4):
    day = "Thursday."
  elif (r == 5):
    day = "Friday."
  elif (r == 6):
    day = "Saturday"
  else:
    day = "Sunday."
  print("The day is",day)
main()

File: Exercise312/src/Day.java
//Import the utility package from java library
import java.util.*;

public class Day
{
  public static void main (String[] args)
  {
    //Create Scanner object
    Scanner input = new Scanner(System.in);

    //Create variable year for year entered and check condition
    System.out.print("Enter year: ");
    int year = input.nextInt(); 
    while (year < 1900 || year > 2100)
    {
     System.out.print("Enter year: ");
     year = input.nextInt();
      
    }

    //Create variable mon for month entered and check condition
    System.out.print("Enter month: ");
    int mon = input.nextInt();
    while (mon > 12 || mon < 1)
    {
      System.out.print("Enter month: "); 
      mon = input.nextInt();
    }

    //Create variable dayNum for day entered and check condition
    System.out.print("Enter day: ");
    int dayNum = input.nextInt();
    while (dayNum < 1 && dayNum > 31)
    {
      System.out.print("Enter day: ");
      dayNum = input.nextInt();
    }

    //If the month is February
    if (mon == 2)
    {
      
      //If the year is a leap year
      if ((year / 100 == 0 && year / 400 == 0) || year / 4 == 0)
      {
        while (dayNum > 28 )
        {
          System.out.print("Enter day: ");
          dayNum = input.nextInt();
        }
      }
      
      // If the year is not a leap year
      else
      {
        while (dayNum > 29 )
          {
            System.out.print("Enter day: ");
            dayNum = input.nextInt();
          }
      }
    }
    
    //If the month is 1, 3, 5, 7, 8, 10, 12
    else if (mon == 1 || mon == 3 || mon == 5 || mon == 7 || mon == 8 || mon == 10 || mon 

== 12)
    {
      while (dayNum > 31 )
      {
        System.out.print("Enter day: ");
        dayNum = input.nextInt();
      }
    }
    
    //If the month is 4, 6, 9, 11
    else if (mon == 4 || mon == 6 || mon == 9 || mon == 11)
      {
        while (dayNum > 30)
        {
          System.out.print("Enter day: ");
          dayNum = input.nextInt();
        }
      }

    //Declare a new variable for month
    int newMon;
    if (mon >= 3)
    {
      newMon = mon - 2;
    }
    else
    {
      newMon = mon + 10; 
    }

    //Declare century
    int century = year / 100;


    //Compute the day using Rev. Zeller's method
    String day = ComputeDay(newMon, dayNum, year, century);

    //Display resutl
    System.out.println("That day was " + day);
  }
  
   public static String ComputeDay(int a, int b, int c, int d)
  {
	String str = null;
    int w = (13 * a - 1) / 5;
    int x = c / 4;
    int y = d / 4;
    int z = w + x + y + b + c - 2 * d;
    int r = z % 7;
    r = (r + 7 ) % 7;
    System.out.print(r);
    switch (r)
    {
      case 0: str = "Sunday"; break;
      case 1: str =  "Monday"; break;
      case 2: str =  "Tuesday"; break;
      case 3: str =  "Wednesday"; break;
      case 4: str =  "Thursday"; break;
      case 5: str =  "Friday"; break;
      case 6: str =  "Saturday"; break;
    }
    return str;
  }
}